google-cloud-datastore/src/main/java/com/google/cloud/datastore/AggregationResults.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.Timestamp;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+public class AggregationResults implements Iterable<AggregationResult> {
+
+  private final List<AggregationResult> aggregationResults;
+  private Timestamp readTime;
+
+  public AggregationResults(List<AggregationResult> aggregationResults,
```

Comment:
As its public constructor we may have to check for null condition for aggregationResults else all method may fail with unexpected behavior.

Reply:
Fixed as part of https://github.com/googleapis/java-datastore/pull/823/commits/6508ff6dc97171ee652a8c9b189f7425e9d4bd45.